assets/quick-order-list.js

```diff
@@ -140,12 +140,32 @@ class QuickOrderList extends HTMLElement {
 
     const quantity = inputValue - cartQuantity;
 
-    if (cartQuantity > 0) {
+    if (inputValue == 0) {
       this.updateQuantity(index, inputValue, name, this.actions.update);
     } else {
-      this.updateQuantity(index, quantity, name, this.actions.add);
+      this.validateQuantity(event, name, index, inputValue, cartQuantity, quantity);
     }
   }
+  
+  validateQuantity(event, name, index, inputValue, cartQuantity) {
+    if (inputValue < event.target.dataset.min) {
+      event.target.setCustomValidity(`This item has a min of ${event.target.dataset.min}`);
+      event.target.reportValidity();
+    } else if (inputValue > parseInt(event.target.max)) {
+      event.target.setCustomValidity(`This item has a max of ${event.target.max}`);
+      event.target.reportValidity();
+    } else if (inputValue % parseInt(event.target.step) != 0) {
+      event.target.setCustomValidity(`This item has an increment of ${event.target.step}`);
+      event.target.reportValidity();
+    } else {
+      event.target.setCustomValidity('');
+      if (cartQuantity > 0) {
+        this.updateQuantity(index, inputValue, name, this.actions.update);
+      } else {
+        this.updateQuantity(index, quantity, name, this.actions.add);
```

Comment:
It's not updating qty when you input the right qty 👇

<img width="784" alt="Screenshot 2024-03-19 at 10 26 52 AM" src="https://github.com/Shopify/dawn/assets/105315663/b9605c8b-9d4c-40a8-a3b3-73f59b09ca0a">

Reply:
Fixed the typo! good catch